portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,9 +41,23 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
+    // Convert the input into an int.
+    String maxCommentsString = request.getParameter("max-comments");
+    int maxComments = 3; // 3 is the default value.
+    try {
+      maxComments = Integer.parseInt(maxCommentsString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + maxCommentsString);
+    }
+
     List<String> comments = new ArrayList<>();
+    int counter = 0;
     for (Entity entity : results.asIterable()) {
+      if (counter >= maxComments) {
+          break;
+      }
       comments.add((String) entity.getProperty("title"));
+      counter++;
     }
```

Comment:
[Optional] I think this is an ok way to implement this but I usually prefer to avoid 'break' statements if possible.  It might be better to just loop based on the counter and extract the entity inside the loop

for (let i = 0; i < maxComments; i++) {
  ...
}

Reply:
I am getting errors. when i try index results or a results.asIterable() object. For now I'll keep the loop as is, but I'll ask some questions about it.

Reply:
I think for an iterable you have to use next().  You can't index.